src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.utils.ts

```diff
@@ -56,3 +68,192 @@ export const getIdentifiers = (
     [],
   );
 };
+
+export type FormattedResponse = {
+  activityItem: Item;
+  answers: ItemAnswer[];
+};
+
+const getAnswers = (currentAnswer: ActivityItemAnswer, date: Date | string) => {
+  switch (currentAnswer.activityItem.responseType) {
+    case ItemResponseType.MultipleSelection:
+      return (currentAnswer.answer as DecryptedMultiSelectionAnswer).value.map((value) => ({
+        answer: {
+          value,
+          text: null,
+        },
+        date,
+      }));
+    default:
+      return [
+        {
+          answer: currentAnswer.answer,
+          date,
+        },
+      ];
+  }
+};
+
+const compareActivityItemAnswers = (
+  prevActivityItemAnswer: FormattedResponse,
+  currActivityItemAnswer: ActivityItemAnswer,
+  date: string,
+) => {
+  const { activityItem: prevActivityItem, answers: prevAnswers } = prevActivityItemAnswer;
+  const { activityItem: currActivityItem, answer: currAnswer } = currActivityItemAnswer;
+
+  if (currActivityItem.responseType === ItemResponseType.SingleSelection) {
+    const currActivityItemOptions = (
+      currActivityItem.responseValues as SingleAndMultipleSelectItemResponseValues
+    ).options;
+    const prevActivityItemOptions = (
+      prevActivityItem.responseValues as SingleAndMultipleSelectItemResponseValues
+    ).options.reduce((acc, cur) => ({ ...acc, [cur.id]: cur }), {});
+
+    const options = currActivityItemOptions.reduce(
+      (
+        acc: Record<string, SingleAndMultipleSelectionOption>,
+        option: SingleAndMultipleSelectionOption,
+      ) => ({ ...acc, [option.id]: option }),
+      prevActivityItemOptions,
+    );
+
+    return {
+      activityItem: {
+        ...currActivityItem,
+        responseValues: {
+          ...currActivityItem.responseValues,
+          options: Object.values(options),
+        },
+      },
+      answers: [
+        ...prevAnswers,
+        {
+          answer: currAnswer,
+          date,
+        },
+      ],
+    };
+  }
+
+  if (currActivityItem.responseType === ItemResponseType.MultipleSelection) {
+    const currActivityItemOptions = (
```

Comment:
Per my understanding, these lines 140-153 are the same with 106-119. Could you please move duplicated code in a function if applicable?